contrib/platform/src/com/sun/jna/platform/mac/IOKit.java

```diff
@@ -0,0 +1,418 @@
+     * @param options
+     *            No options are currently defined.
+     * @param bsdName
+     *            The BSD name.
+     * @return The matching dictionary created, is returned on success, or
+     *         {@code null} on failure.
+     *         <p>
+     *         The dictionary is commonly passed to
+     *         {@link #IOServiceGetMatchingServices} which will consume a reference,
+     *         otherwise it should be released with {@link CoreFoundation#CFRelease}
+     *         by the caller.
+     */
+    CFMutableDictionaryRef IOBSDNameMatching(long masterPort, int options, String bsdName);
+
+    /**
+     * Look up a registered IOService object that matches a matching dictionary.
+     *
+     * @param masterPort
+     *            The master port obtained from {@link #IOMasterPort}.
+     * @param matchingDictionary
+     *            A CF dictionary containing matching information, of which one
+     *            reference is always consumed by this function. IOKitLib can
+     *            construct matching dictionaries for common criteria with helper
+     *            functions such as {@link #IOServiceMatching},
+     *            {@link #IOServiceNameMatching}, and {@link #IOBSDNameMatching}.
+     * @return The first service matched is returned on success.
+     *         <p>
+     *         The service must be released by the caller.
+     */
+    long IOServiceGetMatchingService(long masterPort, CFMutableDictionaryRef matchingDictionary);
+
+    /**
+     * Look up registered IOService objects that match a matching dictionary.
+     *
+     * @param masterPort
+     *            The master port obtained from {@link #IOMasterPort}.
+     * @param matchingDictionary
+     *            A CF dictionary containing matching information, of which one
+     *            reference is always consumed by this function. IOKitLib can
+     *            construct matching dictionaries for common criteria with helper
+     *            functions such as {@link #IOServiceMatching},
+     *            {@link #IOServiceNameMatching}, and {@link #IOBSDNameMatching}.
+     * @param existing
+     *            An iterator handle is returned on success, and should be released
+     *            by the caller when the iteration is finished.
+     * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+     */
+    int IOServiceGetMatchingServices(long masterPort, CFMutableDictionaryRef matchingDictionary,
+            LongByReference existing);
+
+    /**
+     * Returns the next object in an iteration.
+     *
+     * @param iterator
+     *            An IOKit iterator handle.
+     * @return If the iterator handle is valid, the next element in the iteration is
+     *         returned, otherwise zero is returned. The element should be released
+     *         by the caller when it is finished.
+     */
+    long IOIteratorNext(long iterator);
```

Comment:
io_object_t is basicly a pointer to an ipc struct - you could think about binding it typed. The same applies to other derivated of mac_port_t

Reply:
In OSHI, I did originally define wrapper classes for MachPort, IOService, and IOConnect but for the use cases here we are mostly just passing around the pointer addresses to the various functions.  I think mapping the whole IPC structure is overkill, though.   Would these just be extensions of `PointerType`?